volume/devicemapper/devicemapper.go

```diff
@@ -76,6 +76,12 @@ func Init(root string, options []string) (volume.Driver, error) {
 	}
 
 	if err := driver.ensureInitialized(); err != nil {
+		switch err.(type) {
+		case devmapper.ThinpoolBaseDeviceVerificationFailedError:
+			// dm.thinpool argument was not provided but driver is still initialized
+			// so we will not error out
+			return driver, nil
```

Comment:
Seems like this case should fall through and call driver.cleanUpSnapshots() below.

Reply:
that's what we're trying to avoid with this change. We don't want the metadata to be modified  if the argument for dm.thinpool was mistakenly left out